src/main/java/com/stripe/net/RequestMetrics.java

```diff
@@ -0,0 +1,21 @@
+package com.stripe.net;
+
+public class RequestMetrics {
+    public String request_id;
+    public long request_duration_ms;
+
+    public class Payload {
+        public RequestMetrics last_request_metrics;
+    }
+
+    public RequestMetrics(String requestId, long requestDurationMS) {
+        this.request_id = requestId;
+        this.request_duration_ms = requestDurationMS;
+    }
+
+    public Payload payload() {
+        Payload p = new Payload();
+        p.last_request_metrics = this;
```

Comment:
I think this will cause a memory leak as you're going to be holding onto an indefinite chain of references, so the GC can never garbage collect the request metrics objects.  It would be better to create a new request metrics payload that is a copy of the current

Reply:
Chatted in person but I'll record the discussion here:

This won't cause a memory leak because `Payload` and `RequestMetrics` are different types, so we aren't creating any chain of `RequestMetrics` objects.

That being said, this pattern of using an inner class was kind of strange, so I pulled it out into `ClientTelemetryPayload`.